include/seastar/util/memory-data-sink.hh

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2025 ScyllaDB
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <seastar/core/iostream.hh>
+
+namespace seastar {
+namespace util {
+
+#if SEASTAR_API_LEVEL >= 9
+
+template <template <typename T> class Container>
+requires requires (Container<temporary_buffer<char>>& col, size_t s) {
+    typename std::back_insert_iterator<Container<temporary_buffer<char>>>;
+}
+inline void append_buffers(Container<temporary_buffer<char>>& c, std::span<temporary_buffer<char>> bufs) {
+    if constexpr (requires(decltype(c), size_t s) { c.reserve(s); }) {
+        c.reserve(c.size() + bufs.size());
```

Comment:
This is quadratic, let's say bufs.size() is constant 1, each time we'll reallocate the entire vector.